lib/rules/require-expect.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @fileoverview Require the use of `expect` when using `assert` inside of a
+ * block or when passing `assert` to a function.
+ * @author Mitch Lloyd
+ */
+"use strict";
+
+var utils = require("../utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    var EXCEPT_SIMPLE_ERROR_MESSAGE = "Should use `assert.expect` when using assertions outside of the top-level test callback",
+        ALWAYS_ERROR_MESSAGE = "Test is missing `assert.expect() call`",
+        currentTest = false;
+
+    function isExpectCall(callee) {
+        return callee.object &&
+               callee.object.type === "Identifier" &&
+               callee.object.name === currentTest.assertName &&
+               callee.property.name === "expect";
```

Comment:
Yeah, I want to stay consistent with the rest of the rules in this library so I'll add that.